libr/core/cmd_anal.c

```diff
@@ -1108,27 +1113,26 @@ static int cmd_anal_fcn(RCore *core, const char *input) {
 			r_core_cmd_help (core, help_afC);
 			} break;
 		case 'l':{ //afCl list all function Calling conventions.
-			r_cons_println (r_anal_cc_type2str (R_ANAL_CC_TYPE_CDECL));
-			r_cons_println (r_anal_cc_type2str (R_ANAL_CC_TYPE_STDCALL));
-			r_cons_println (r_anal_cc_type2str (R_ANAL_CC_TYPE_FASTCALL));
-			//THOSE are the only implemented ones
-			//should I test for null ... no ;)
+
+				sdb_foreach (core->anal->sdb_cc, cc_print, NULL);
 			} break;
 		case 'a':
 			eprintf ("Todo\n");
 			break;
 		case ' ': {
-			int type = r_anal_cc_str2type (input + 3);
-			if (type == -1) {
+			char *cc = r_str_chop (strdup (input + 3));
+			if (!r_anal_cc_exist (core->anal, cc)) {
 				eprintf ("Unknown calling convention '%s'\n"
-					"See afCl for available types\n", input + 3);
+					"See afCl for available types\n", cc);
 			} else {
-				fcn->call = type;
+				if (fcn->cc) {
+					free (fcn->cc);
```

Comment:
free does for you so is not needed `if (fcn->cc)`. From glibc source code for example

```C
void __libc_free (void *mem) {
...
if (mem == 0)                              /* free(0) has no effect */
    return;
```